python/ray/data/dataset.py

```diff
@@ -362,11 +362,13 @@ def _move_blocks(self):
         self._blocks.clear()
         return blocks
 
-    def split(self,
-              n: int,
-              *,
-              equal: bool = False,
-              locality_hints: List[Any] = None) -> List["Dataset[T]"]:
+    def split(
+            self,
+            n: int,
+            *,
+            equal: bool = False,
+            blocks_per_split: Optional[int] = None,
```

Comment:
Can we remove this feature? It wasn't in the original specification of the split API.

Reply:
The blocks within a split shard can get skewed due to block splitting, which can be pretty harmful to downstream computations (e.g. one map task getting a 1 row block while another gets a 10 million row block). I suppose that we can automatically rebalance the blocks on a best-effort basis via choosing some sensible number of blocks, such as we currently do when no `blocks_per_split` argument is given, and look at exposing this parameter in the future if any user ends up wanting more control.